### DISCUSSION TITLE
pip install timbral_models  error

### DISCUSSION THREAD

**Discussion by Shamit0906** [community/community#110287] [Programming Help]
### Select Topic Area

Question

### Body

Collecting timbral_models
  Using cached timbral_models-0.4.0.tar.gz (59 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Requirement already satisfied: numpy in c:\users\cse\anaconda3\envs\speech\lib\site-packages (from timbral_models) (1.26.4)
Requirement already satisfied: soundfile in c:\users\cse\anaconda3\envs\speech\lib\site-packages (from timbral_models) (0.12.1)
Requirement already satisfied: scipy in c:\users\cse\anaconda3\envs\speech\lib\site-packages (from timbral_models) (1.12.0)
Requirement already satisfied: librosa in c:\users\cse\anaconda3\envs\speech\lib\site-packages (from timbral_models) (0.10.1)
Collecting sklearn (from timbral_models)
  Using cached sklearn-0.0.post12.tar.gz (2.6 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
Note: you may need to restart the kernel to use updated packages.
  error: subprocess-exited-with-error
  
  python setup.py egg_info did not run successfully.
  exit code: 1
  
  [15 lines of output]
  The 'sklearn' PyPI package is deprecated, use 'scikit-learn'
  rather than 'sklearn' for pip commands.
  
  Here is how to fix this error in the main use cases:
  - use 'pip install scikit-learn' rather than 'pip install sklearn'
  - replace 'sklearn' by 'scikit-learn' in your pip requirements files
    (requirements.txt, setup.py, setup.cfg, Pipfile, etc ...)
  - if the 'sklearn' package is used by one of your dependencies,
    it would be great if you take some time to track which package uses
    'sklearn' instead of 'scikit-learn' and report it to their issue tracker
  - as a last resort, set the environment variable
    SKLEARN_ALLOW_DEPRECATED_SKLEARN_PACKAGE_INSTALL=True to avoid this error
  
  More information is available at
  https://github.com/scikit-learn/sklearn-pypi-package
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

Encountered error while generating package metadata.

See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

**How to Resolve**

**Comment by ghost** [comment]
First run this following command,

```python
pip install scikit-learn
```

Then try to install timbral_models

**Reply by Shamit0906** [reply]
I tried but not working

**Reply by ghost** [reply]
Then try following this Documentation, i Guess this would help,

[Documentation](https://github.com/scikit-learn/sklearn-pypi-package)

**Reply by ghost** [reply]
Let me just clarify this to you, 
the package you are trying to install has a requirements.txt file where it has a dependency mentioned in it that is sklearn, but it has been renamed as scikit-learn. This is what causing the issue

**Reply by Shamit0906** [reply]
Okk

**Comment by Shamit0906** [comment]
It asked about to change in timbral_models.config file from sklern to scikit , but how can't understand

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @Shamit0906 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
